app/scripts/migrations/125.ts

```diff
@@ -0,0 +1,50 @@
+import { cloneDeep } from 'lodash';
+import { hasProperty, isObject } from '@metamask/utils';
+
+type VersionedData = {
+  meta: { version: number };
+  data: Record<string, unknown>;
+};
+
+export const version = 125;
+
+/**
+ * This migration removes any dangling instances of SelectedNetworkController.perDomainNetwork
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(
+  originalVersionedData: VersionedData,
+): Promise<VersionedData> {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  transformState(versionedData.data);
+  return versionedData;
+}
+
+// TODO: Replace `any` with specific type
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
We should use `unknown` here rather than `any`, `any` is disallowed.

See https://github.com/MetaMask/contributor-docs/blob/main/docs/typescript.md#avoid-any

Reply:
done here https://github.com/MetaMask/metamask-extension/pull/26265/commits/28414ebceccbabf3bea0032ef095ac5b86d9cb52